stats/thresholds_test.go

```diff
@@ -56,9 +54,8 @@ func TestNewThreshold_InvalidThresholdConditionExpression(t *testing.T) {
 	abortOnFail := false
 	gracePeriod := types.NullDurationFrom(2 * time.Second)
 
-	gotThreshold, err := newThreshold(src, abortOnFail, gracePeriod)
+	gotThreshold := newThreshold(src, abortOnFail, gracePeriod)
 
-	assert.Error(t, err, "instantiating a threshold with an invalid expression should fail")
```

Comment:
Can you instead of removing this, call Parse and check the error from it?

Reply:
To clarify, do you mean calling `Parse` in the test here, or calling it in `UnmarshalJSON`?

PS: calling it inside the test would be equivalent to test `Parse` itself. But I want to make sure :)

Reply:
I meant calling it here on its own, so you can just get the same error you got before. 

This also is true for all other places – just everywhere call `Parse` after the thresholds are created and check the error

Reply:
Alright, will do 👍🏻

Reply:
I actually had forgotten to remove `parseThresholdExpression` call from `newThreshold`. It's now done. You did in your patch too. It makes sense.

Following that, I don't think it makes much sense to call `Parse` in the `newThreshold` (singular) tests anymore, as it would only test the input we give in the test is parsable, and would not assert anything about `newThreshold` itself anymore. Would you agree?

Reply:
On a similar note the current test failures are because you don't Parse it in some cases where that is needed as it will then be used. 

Given that this in practice just restores to what hte test were before I feel that is much better either way.

Reply:
Indeed, I caught that in the process. Thanks for pointing it out anyway 😸 I've added parsing steps to the tests.